crates/uv-publish/src/trusted_publishing.rs

```diff
@@ -36,15 +32,10 @@ pub enum TrustedPublishingError {
     /// When trusted publishing is misconfigured, the error above should occur, not this one.
     #[error("PyPI returned error code {0}, and the OIDC has an unexpected format.\nResponse: {1}")]
     InvalidOidcToken(StatusCode, String),
-}
-
-impl TrustedPublishingError {
-    fn from_var_err(env_var: &'static str, err: VarError) -> Self {
-        match err {
-            VarError::NotPresent => Self::MissingEnvVar(env_var),
-            VarError::NotUnicode(os_string) => Self::InvalidEnvVar(env_var, os_string),
-        }
-    }
+    /// This error can only occur when the user forces the use of trusted publishing
+    /// in an unsupported environment.
+    #[error("No OIDC token available from the environment")]
```

Comment:
Can we add some specific suggestions about what's gone wrong? Does this only happen when running outside of CI or also when GitLab CI is misconfigured?

Reply:
Yeah, I'll add suggestions here. This specifically happens when `ambient_id`'s discovery returns `None`, which happens if none of the discovery methods match. That roughly corresponds to running outside of CI, since discovery will fail (instead of returning `None`) if the environment has `GITHUB_CI` or `GITLAB_CI` but none of the other required state.